pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1108,11 +1114,17 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
 			tlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs = append(tlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs,
 				authn_model.ConstructSdsSecretConfig(res.GetResourceName(), node.RequestedTypes.CDS))
 
-			// If tls.CaCertificate or CaCertificate in Metadata isn't configured don't set up RootSdsSecretConfig
-			if res.GetRootResourceName() == "" {
+			if res.GetRootResourceName() == "" && features.EnableDefaultToSystemCAValidation.Get() {
+				res = model.SdsCertificateConfig{
+					CaCertificatePath: features.SystemCAFilePath.Get(),
```

Comment:
Same - you lost SNI. 

And we also need to add the SAN - root verification without SAN check is not very useful.

Re. SNI: the problem is that without sending the SNI, many hosts will not return the right certificate.

Reply:
Unless Envoy sends the right SNI by default - I haven't checked.

Reply:
Sni is set before this logic when tlscontext is inutialized

Reply:
theres unit tests too

Reply:
https://github.com/istio/istio/blob/4e7294303ac89e71c4d718220162f616d3a57867/pilot/pkg/networking/core/v1alpha3/cluster.go#L1008

Reply:
https://github.com/istio/istio/blob/4e7294303ac89e71c4d718220162f616d3a57867/pilot/pkg/networking/core/v1alpha3/cluster.go#L1010